src/server/pfs/server/driver.go

```diff
@@ -157,8 +157,57 @@ func newDriver(env *serviceenv.ServiceEnv, etcdPrefix string, treeCache *hashtre
 	return d, nil
 }
 
+// Operation is an interface for specifying multiple requests to be performed
+// transactionally through 'performRequests'
+type Operation interface {
+	validate(d *driver, pachClient *client.APIClient) error
+	execute(d *driver, pachClient *client.APIClient, stm col.STM) error
+}
+
+// CreateBranchRequest is a type alias for the protobuf pfs.CreateBranchRequest
+type CreateBranchRequest pfs.CreateBranchRequest
+
+func (d *driver) performRequest(pachClient *client.APIClient, request Operation) error {
```

Comment:
Could we call this `performOperation` or maybe even `runOp` to be really terse. I like having a distinction between a `request` as a dumb data structure that lives in proto land and an `Operation` as a smarter interface that lives in driver, but we should be consistent with the naming.

Reply:
Heh, I've already renamed this in the next PR - it's currently `executeRequests`, but I think `executeTransaction`/`runTransaction` should make it very clear - do either of those sound good?

I like `execute` because it lines up with the `Operation.execute` method, especially now that `validate` has been rolled into `execute` (in the follow-up PR).

Reply:
Also, I've renamed `execute` to `executeInTransaction` so that Requests can expose an `execute` method that runs in its own transaction and returns the result directly instead of through a serialized protobuf message (for ease of use/readability if a batch transaction is unnecessary).

Reply:
`executeTransaction` makes sense to me, it would be weird if it ran both a `validate` method and an `execute` method, but if you've slimmed it down to just an `execute` method then I think the name fits perfectly. I think I have a slight preference for `run` over `execute` since it's shorter and we sometimes use `Txn` as shorthand for `Transaction`. I'm happy with any of these options, they're all descriptive enough and sufficiently terse.